modules/unsupported/geometry/src/main/java/org/geotools/geometry/iso/sfcgal/util/Geometry3DOperation.java

```diff
@@ -0,0 +1,257 @@
+         */
+        public static boolean disjoint(GeometryImpl gA, GeometryImpl gB) {
+                SFGeometry geometryA = SFCGALConvertor.geometryToSFCGALGeometry(gA);
+                SFGeometry geometryB = SFCGALConvertor.geometryToSFCGALGeometry(gB);
+
+                return !SFAlgorithm.intersects3D(geometryA, geometryB);
+        }
+
+        
+        /**
+         * @param gA
+         * @param gB
+         * @return TRUE, if the gA is intersect with gB
+         */
+        public static boolean intersects(GeometryImpl gA, GeometryImpl gB) {
+                SFGeometry geometryA = SFCGALConvertor.geometryToSFCGALGeometry(gA);
+                SFGeometry geometryB = SFCGALConvertor.geometryToSFCGALGeometry(gB);
+
+                return SFAlgorithm.intersects3D(geometryA, geometryB);
+        }
+
+        /**
+         * @param gA
+         * @param gB
+         * @return TRUE, if the gA touches gB.
+         */
+        public static boolean touches(GeometryImpl gA, GeometryImpl gB) {
+                IntersectionMatrix3D tIM = null;
+                try {
+                        tIM = RelateOp3D.relate(gA, gB);
+                } catch (Exception e) {
+                        e.printStackTrace();
+                }
+
+                boolean rValue = false;
+                if (tIM.matches("FT**") || tIM.matches("F*T*") || tIM.matches("F**T")) {
+                        rValue = true;
+                }
+
+                return rValue;
+        }
+
+        /**
+         * @param gA
+         * @param gB
+         * @return TRUE, if the gA is spatially contain gB.
+         */
+        public static boolean contains(GeometryImpl gA, GeometryImpl gB) {
+                SFGeometry geometryA = SFCGALConvertor.geometryToSFCGALGeometry(gA);
+                SFGeometry geometryB = SFCGALConvertor.geometryToSFCGALGeometry(gB);
+
+                return !geometryA.equals(geometryB) && !touches(gA, gB)
+                                && SFAlgorithm.covers3D(geometryA, geometryB);
+        }
+
+        /**
+         * @param gA
+         * @param gB
+         * @return TRUE, if the gA is spatially within gB.
+         */
+        public static boolean within(GeometryImpl gA, GeometryImpl gB) {
+                return contains(gB, gA);
+        }
+
+        /**
+         * @param gA
+         * @param gB
+         * @return TRUE, if the gA overlaps with gB.
+         */
+        public static boolean overlaps(GeometryImpl gA, GeometryImpl gB) {
+                SFGeometry geometryA = SFCGALConvertor.geometryToSFCGALGeometry(gA);
+                SFGeometry geometryB = SFCGALConvertor.geometryToSFCGALGeometry(gB);
+                SFGeometry intersection = SFAlgorithm.intersection3D(geometryA, geometryB);
+                boolean rValue = false;
+
+                if (equals(gA, gB))
+                        return rValue;
+                if (!intersection.isEmpty() && !contains(gA, gB) && !contains(gB, gA)

```

Comment:
These function calls are inefficient. You need to avoid duplication of operations.